src/js/editor/post.js

```diff
@@ -602,20 +602,23 @@ class PostEditor {
    * or, if a string, the tag name of the markup (e.g. 'strong', 'em') to toggle.
    */
   toggleMarkup(markupOrMarkupString) {
-    const markup = typeof markupOrMarkupString === 'string' ?
-                     this.builder.createMarkup(markupOrMarkupString) :
-                     markupOrMarkupString;
-
-    const range = this.editor.cursor.offsets;
-    const hasMarkup = m => m.hasTag(markup.tagName);
-    const rangeHasMarkup = any(this.editor.markupsInSelection, hasMarkup);
-
-    if (rangeHasMarkup) {
-      this.removeMarkupFromRange(range, hasMarkup);
-    } else {
-      this.applyMarkupToRange(range, markup);
+    if (this.editor.cursor.hasSelection()) {
```

Comment:
can this be refactored to not need to know if there is a selection?
`removeMarkupFromRange` and `applyMarkupToRange` should be safe to run on a collapsed range.